client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -14,7 +23,36 @@ const PatternAssembler: Step = ( { navigation } ) => {
 	const [ footer, setFooter ] = useState< Pattern | null >( null );
 	const [ sections, setSections ] = useState< Pattern[] >( [] );
 	const [ sectionPosition, setSectionPosition ] = useState< number | null >( null );
-	const { goNext, goBack } = navigation;
+	const { goBack, goNext, submit } = navigation;
+	const { setDesignOnSite } = useDispatch( SITE_STORE );
+	const reduxDispatch = useReduxDispatch();
+	const { setPendingAction } = useDispatch( ONBOARD_STORE );
+	const selectedDesign = useSelect( ( select ) => select( ONBOARD_STORE ).getSelectedDesign() );
+	const site = useSite();
+	const siteSlug = useSiteSlugParam();
+	const siteId = useSiteIdParam();
+	const siteSlugOrId = siteSlug ? siteSlug : siteId;
+
+	const getDesign = () =>
+		( {
+			...selectedDesign,
+			recipe: {
+				...selectedDesign?.recipe,
+				header_pattern_ids: header ? [ encodePatternId( header.id ) ] : undefined,
+				pattern_ids: sections.filter( Boolean ).map( ( pattern ) => encodePatternId( pattern.id ) ),
+				footer_pattern_ids: footer ? [ encodePatternId( footer.id ) ] : undefined,
+			} as DesignRecipe,
+		} as Design );
+
+	const getPageTemplate = () => {
+		let pageTemplate = 'footer-only';
+
+		if ( header ) pageTemplate = 'header-footer-only';
+		if ( ! header && footer ) pageTemplate = 'footer-only';
+		if ( ! header && ! footer && sections.length ) pageTemplate = 'blank';
```

Comment:
[nit] I personally don't like one line if-condition. Just express my thoughts 😅

Reply:
I don't like switch and semicolons 😅